### PR TITLE
c-CATch 5.0: new recipe

### PR DESCRIPTION
**Pull request by TheZeldakatze**
I quickly ported this game when trying to port another game and then also used it to understand how the packaging process works. The game itself is pretty simple.

### CODE REVIEW

File: games-arcade/ccatch/ccatch-5.0.recipe [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+HOMEPAGE="http://www.victorswelt.de/catch.html"
+COPYRIGHT="2020 Victor Gamper"
+LICENSE="GNU GPL v3"
+REVISION="1"
+srcGitRev="018c0fd7184e02992488a95ce514c3f1d77d7604"
+SOURCE_URI="https://github.com/TheZeldakatze/c-CATch/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="c5f0d7b696d641b6168b92b05e822f76c92de31c784739d94f901b5596e593c5"
+SOURCE_FILENAME="c-CATch-$srcGitRev.tar.gz"
+SOURCE_DIR="c-CATch-$srcGitRev"
+# 6 Dec 2021 commit.
```
Comment (Begasus) [inline games-arcade/ccatch/ccatch-5.0.recipe:12]: New line between SOURCE_DIR and ARCHITECTURES (https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering) , maybe move the commit line before srcGitRev?

File: games-arcade/ccatch/ccatch-5.0.recipe [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+	app:cCATch = $portVersion
+	"
+REQUIRES="
+	haiku
+	lib:libsdl$secondaryArchSuffix
+        "
+
+BUILD_REQUIRES="
+	haiku_devel
+	devel:libsdl$secondaryArchSuffix
```
Comment (Begasus) [inline games-arcade/ccatch/ccatch-5.0.recipe:26]: This should be `libSDL_1.2` (in both cases) as mentioned in: https://github.com/haikuports/haikuports/issues/3878

File: games-arcade/ccatch/ccatch-5.0.recipe [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+{
+	gcc -lSDL -o c-CATch catch.c BitmapFontRenderer.c PidginImage.c
+}
+
+INSTALL()
+{
+	mkdir -p $appsDir
+	cp c-CATch $appsDir/cCATch
+	addAppDeskbarSymlink $appsDir/cCATch
+}
```
Comment (Begasus) [inline games-arcade/ccatch/ccatch-5.0.recipe:43]: New line at the end.

**Review by Begasus — CHANGES_REQUESTED** [review]

File: games-arcade/ccatch/ccatch-5.0.recipe (hunk 1 above)
Comment (TheZeldakatze) [reply games-arcade/ccatch/ccatch-5.0.recipe]: Fixed that, I wonder how I managed to put the commit line at the bottom

**Comment by TheZeldakatze** [comment]
is the second commit bad in a way? I somehow managed to pull from haikuports into the new-feature branch, is that a problem and if so, how can I revert that?

File: games-arcade/ccatch/ccatch-5.0.recipe [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ARCHITECTURES="all"
+
+PROVIDES="
+	cCATch = $portVersion
+	app:cCATch = $portVersion
+	"
+REQUIRES="
+	haiku
+	lib:libSDL_1.2$secondaryArchSuffix
+        "
```
Comment (Begasus) [inline games-arcade/ccatch/ccatch-5.0.recipe:24]: One tab to many (saw you used spaces here, always use tab(s)):
```diff
-         "
+ 	"
```

**Comment by Begasus** [comment]
> is the second commit bad in a way? I somehow managed to pull from haikuports into the new-feature branch, is that a problem and if so, how can I revert that?

I don't see any issues here, no other stuff is involved, the branch looks clean to me.
Didn't have time yet to checkout a build, will report later (or someone else if they can) on the build, other then that it looks ok to me.

File: games-arcade/ccatch/ccatch-5.0.recipe [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+ARCHITECTURES="all"
+
+PROVIDES="
+	cCATch = $portVersion
+	app:cCATch = $portVersion
+	"
+REQUIRES="
+	haiku
+	lib:libSDL_1.2$secondaryArchSuffix
```
Comment (Begasus) [inline games-arcade/ccatch/ccatch-5.0.recipe:23]: No need for `$secondaryArchSuffix` if this builds with `gcc2` (further down also), if not it should be added to other places as well and `x86_gcc2` should be marked broken (`!x86_gcc2`) in `ARCHITECTURES`.

File: games-arcade/ccatch/ccatch-5.0.recipe [hunk 6]
```diff
@@ -0,0 +1,46 @@
+SUMMARY="Help a cat catch birds"
+DESCRIPTION="cCATch is a portation of the game CATch, written in c using SDL"
+HOMEPAGE="https://www.github.com/TheZeldakatze/c-CATch"
+COPYRIGHT="2020 Victor Gamper"
+LICENSE="GNU GPL v3"
+REVISION="2"
```
Comment (Begasus) [inline games-arcade/ccatch/ccatch-5.0.recipe:6]: This is still a new recipe so `REVISION` should stay at "`1`" ;)

**Comment by Begasus** [comment]
Not really needed, but there are variables you could use for your rdef file (renamed to rdef.in in most cases), see: https://github.com/haikuports/haikuports/blob/4f180bdb94db9749ae800820b271e262a12754aa/app-misc/abstrakt/abstrakt-0.0.1%7E20180515.recipe#L66
I also see you are using major=5 middle=0 and minor=0, but your recipe mentiones 5.0, shouldn't that be 5.0.0 then?
Some thing to consider is using ~git.recipe, in the example here you can see how it can cut the version for the rdef.in file then.
For future updates it would reduce the changes needed to update the recipe. :)

File: games-arcade/ccatch/ccatch-5.0.recipe (hunk 5 above)
Comment (TheZeldakatze) [reply games-arcade/ccatch/ccatch-5.0.recipe:23]: I tried to compile it using x86_gcc2 in a virtual machine. It didn't compile as it is using c99.
So I changed the ARCHITECTURES to:
`ARCHITECTURES="all !x86_gcc2"
SECONDARY_ARCHITECTURES="x86"`

How do I test the x86_32 variant?

Comment (Begasus) [reply games-arcade/ccatch/ccatch-5.0.recipe:23]: You build it with `hp ccatch_x86` then ...

Comment (Begasus) [reply games-arcade/ccatch/ccatch-5.0.recipe:23]: Did a quick check, doesn't build for gcc2 (parse errors), so best to stick with `SECONDARY_ARCHITECTURES="x86"` then :)

**Comment by TheZeldakatze** [comment]
ok, I hope I cleaned up the recipe correctly. Sorry for making all those mistakes, I'm new to porting to Haiku

File: games-arcade/ccatch/ccatch-5.0.0.recipe [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+CHECKSUM_SHA256="c5f0d7b696d641b6168b92b05e822f76c92de31c784739d94f901b5596e593c5"
+SOURCE_FILENAME="c-CATch-$srcGitRev.tar.gz"
+SOURCE_DIR="c-CATch-$srcGitRev"
+ADDITIONAL_FILES="ccatch.rdef.in"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	cCATch = $portVersion
```
Comment (Begasus) [inline games-arcade/ccatch/ccatch-5.0.0.recipe:19]: PROVIDES should match the recipe name (you can check this if you are running `haikuporter -S`), also missing $secondaryArchSuffix
```diff
- 	cCATch = $portVersion
+ 	ccatch$secondaryArchSuffix = $portVersion
```

File: games-arcade/ccatch/ccatch-5.0.0.recipe [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	app:cCATch = $portVersion
+	"
+REQUIRES="
+	haiku
```
Comment (Begasus) [inline games-arcade/ccatch/ccatch-5.0.0.recipe:23]: Missing $secondaryArchSuffix:
```diff
- 	haiku
+ 	haiku$secondaryArchSuffix
```

File: games-arcade/ccatch/ccatch-5.0.0.recipe [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+	lib:libSDL_1.2$secondaryArchSuffix
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
```
Comment (Begasus) [inline games-arcade/ccatch/ccatch-5.0.0.recipe:28]: Missing $secondaryArchSuffix:
```diff
- 	haiku_devel
+ 	haiku${secondaryArchSuffix}_devel
```

File: games-arcade/ccatch/ccatch-5.0.0.recipe [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+{
+	gcc -lSDL -o c-CATch catch.c BitmapFontRenderer.c PidginImage.c
+}
+
+INSTALL()
+{	
+	# prepare the rdef file
+	local MAJOR="`echo "$portVersion" | cut -d. -f1`"
+	local MIDDLE="`echo "$portVersion" | cut -d. -f2`"
+	local MINOR="`echo "$portVersion" | cut -d. -f3 | cut -d~ -f1`"
```
Comment (Begasus) [inline games-arcade/ccatch/ccatch-5.0.0.recipe:46]: As you are not using `ccatch-5.0.0~git.recipe`, this isn't needed (not checked atm).
```diff
- 	local MINOR="`echo "$portVersion" | cut -d. -f3 | cut -d~ -f1`"
+ 	local MINOR="`echo "$portVersion" | cut -d. -f3`"
```

**Comment by TheZeldakatze** [comment]
Ok, now everything should be fixed
And I just noticed that I could have just accepted the changes without a new commit. Sorry, it's the first time I did a pull request

**Review by Begasus — APPROVED** [review]

**Comment by Begasus** [comment]
LGTM, didn't check the latest changes in a build though, but looking at the recipe it should be OK :)
Thanks for taking the time to make changes! :+1:

**Comment by TheZeldakatze** [comment]
> LGTM, didn't check the latest changes in a build though, but looking at the recipe it should be OK :)
> Thanks for taking the time to make changes! 👍

And thank you for helping me clean up the recipe

**Pull request merged**